dhall-docs/README.md

```diff
@@ -97,6 +60,377 @@ dhall-docs --input . --output-link ${OTHER_LINK}
 
 For more information about the tool, check the `--help` flag.
 
+
+## Documenting your packages
+
+Documenting your package is essentially writing comments on your source code,
+although there are some format rules that these needs to obey in order to work
+properly with `dhall-docs`. These rules aid `dhall-docs` in extracting the text
+on the documentation that will be passed to our Markdown preprocessor to finally
+render your documentation in HTML.
+
+You can see examples of writing documentation on the [tests](./tasty/data/package)
+folder.
+
+### Documentation markup language
+
+The markup language of the documentation is [CommonMark], which is a strict
+Markdown flavour. `dhall-docs` uses [`mmark`], a package that parses
+[CommonMark] and transforms it to HTML. This package follows (almost)
+that specification.
+
+### Block comments
+
+Normal block comments in Dhall starts with `{-` and ends with `-}`. For example:
+
+```dhall
+{- foo
+     bar
+       baz
+-}
+```
+
+A `dhall-docs` block comment needs to start with `"{-|\n"` i.e. a `{-|` and a
+newline. This means that the actual documentation _must_ start on the following line.
+
+[CommonMark] supports some features that are sensitive to indentation (e.g.
+[indented-code-blocks]). Indentation on `dhall-docs` block comments is really
+similar to [Dhall multi-line strings]: the indentation is determined by the
+longest common whitespace prefix between the lines on the block comment. The
+main difference is that, unlike the closing pair of double single quotes on multi-line
+strings (`''`), the closing brace (`-}`) doesn't alter indentation.
```

Comment:
From my POV it would be annoying if me, a user, should care about the position of the closing brace `-}`

Reply:
Hmm, I'm not convinced that this is important enough that it would be worth breaking the analogy for. There's also at least one case where the significance of the position of the `-}` would be useful:

```
{-
    indented
    code
    block
-}
```

Reply:
Good point :) then a `dhall-docs` block comment will be basically a multi-string :+1: